File: cSharp/day18.cs

using System;


Console.WriteLine("Input a string: ");
string name = Console.ReadLine();

char[] nameChars = name.ToUpper().ToCharArray();

char[] uniqueChars 
    = nameChars.Distinct().ToArray();//.toUnique(*)"??

string uniqueCharacterListString 
    =  String.Join(
    '\n',
    uniqueChars
);

Console.WriteLine(
    uniqueCharacterListString
);

File: go/day18.go
package main
import "strings"

import (
	"fmt"
)

func main() {
	fmt.Println("input text:")
    var name string;
    fmt.Scanln(&name)
	name = strings.ToUpper(name); // case insensitive
	seen := make(map[rune]bool)
	var uniqueChars []rune

	for _, char := range name {
		if !seen[char] {
			seen[char] = true
			uniqueChars = append(uniqueChars, char)
		}
	}

	for _, char := range uniqueChars {
		fmt.Println(string(char))
	}
}
